.github/workflows/cicd.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This is the top-level workflow that runs on each pull request and push to main.
+# It invokes other jobs to perform builds and run tests.
+# All jobs run in parallel, using build artifacts to synchronize jobs.
+
+name: CI/CD
+
+# Run on push so we can capture the baseline code coverage for any squash commit.
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+permissions:
+  contents: read
+  checks: read  # Required by reusable-test.yml to check build status.
+
+jobs:
+  # Perform the regular build.
+  regular:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64
+
+  # Build with C++ static analyzer.
+  analyze:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64-Analyze
+      build_options: /p:Analysis='True'
+
+  # Build with C++ address sanitizer.
+  sanitize:
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      build_artifact: Build-x64-Sanitize
+      build_options: /p:Sanitizer='True'
+
+  # Run the unit tests in GitHub.
+  unit_tests:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      name: unit_tests
+      test_command: unit_tests.exe -s
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: windows-2019
+      code_coverage: true
+      gather_dumps: true
+
+  # Run the API tests in GitHub.
+  ebpf_client:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      name: ebpf_client
+      test_command: ebpf_client.exe -s
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: windows-2019
+      code_coverage: true
+      gather_dumps: true
+
+  # Run the fuzzing tests in GitHub.
+  fuzzing:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      name: fuzzing
+      test_command: fuzz.exe -s
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: windows-2019
+      code_coverage: true
+      gather_dumps: true
+
+  # Run the bpf2c tests in GitHub.
+  bpf2c:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      pre_test: call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat"
+      test_command: bpf2c_tests.exe -s
+      name: bpf2c
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: windows-2019
+      code_coverage: true
+      gather_dumps: true
+
+  # Run the driver tests on self-hosted runners.
+  driver:
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      test_command: powershell ".\setup_ebpf_cicd_tests.ps1; .\execute_ebpf_cicd_tests.ps1; .\cleanup_ebpf_cicd_tests.ps1"
+      name: driver
+      build_job: regular / build
+      build_artifact: Build-x64
+      environment: ebpf_cicd_tests
```

Comment:
code_coverage and gather_dumps should be true for this as well. Note that the tools are already present in the VMs.

Reply:
Yes on the dumps. Will need to do more stuff to get the code coverage working as part of another PR.